infrastructure/sentry-subgraph/src/referee.ts

```diff
@@ -506,40 +543,211 @@ export function handleUnstakeV1(event: UnstakeV1): void {
+    return;
+  }
+
+  //submitAssertionToChallenge = 0xb48985e4
+  //submitMultipleAssertions = 0xec6564bf
+  const transactionSignature = getTxSignatureFromEvent(event)
+
+  let submittedFrom = transactionSignature == "0xb48985e4" ? "submitAssertion" : transactionSignature == "0xec6564bf" ? "submitMultipleAssertions" : "unknown"
+
+  const submission = new Submission(event.params.challengeId.toString() + "_" + event.params.nodeLicenseId.toString())
+  submission.nodeLicenseId = event.params.nodeLicenseId
+  submission.challengeNumber = event.params.challengeId
+  submission.claimed = false
+  submission.claimAmount = BigInt.fromI32(0)
+  submission.sentryKey = event.params.nodeLicenseId.toString()
+  submission.challenge = challenge.id
+  submission.createdTimestamp = event.block.timestamp
+  submission.createdTxHash = event.transaction.hash
+  submission.claimTimestamp = BigInt.fromI32(0)
+  submission.claimTxHash = Bytes.fromI32(0)
+  submission.claimedFrom = "unclaimed"
+  submission.submittedFrom = submittedFrom
+  submission.assertionsStateRootOrConfirmData = event.params.assertionStateRootOrConfirmData.toHexString()
+  submission.eligibleForPayout = event.params.eligibleForPayout
+  submission.save()
+
+  if (submission.eligibleForPayout) {
+    challenge.numberOfEligibleClaimers = challenge.numberOfEligibleClaimers.plus(BigInt.fromI32(1))
+    challenge.save()
+  }
+}
+
+export function handleAssertionCancelled(event: AssertionCancelledEvent): void {
+  // query for the challenge and update it
+  const challenge = Challenge.load(event.params.challengeId.toString())
+
+  if (!challenge) {
+    log.warning("Failed to find challenge handleAssertionCancelled challengeId: " + event.params.challengeId.toString() + ", TX: " + event.transaction.hash.toHexString(), [])
+    return;
+  }
+
+  const submission = Submission.load(event.params.challengeId.toString() + "_" + event.params.nodeLicenseId.toString())
+  if (!submission) {
+    log.warning("Failed to find submission handleAssertionCancelled: nodeLicenseId: " + event.params.nodeLicenseId.toString() + ", challengeId: " + event.params.challengeId.toString() + ", TX: " + event.transaction.hash.toHexString(), [])
+    return;
+  }
+
+  submission.eligibleForPayout = false;
+  submission.save();
+
+  challenge.numberOfEligibleClaimers = challenge.numberOfEligibleClaimers.minus(BigInt.fromI32(1));
+  challenge.save();
+}
+
+
+
+export function handleRewardsClaimedV2(event: RewardsClaimedV2Event): void {
+  // query for the challenge and update it
+  const challenge = Challenge.load(event.params.challengeId.toString())
+
+  if (!challenge) {
+    log.warning("Failed to find challenge handleRewardsClaimedV2 challengeId: " + event.params.challengeId.toString() + ", TX: " + event.transaction.hash.toHexString(), [])
+    return;
+  }
+
+  //claimReward = 0x86bb8f37
+  //claimMultipleRewards = "0xb4d6b7df"
+  const transactionSignature = getTxSignatureFromEvent(event)
```

Comment:
Would adding the claimPoolSubmissionRewards signature here prevent the need for "unknown"?